src/cursor/abstract_cursor.ts

```diff
@@ -309,33 +305,26 @@ export abstract class AbstractCursor<
 
     try {
       while (true) {
+        if (this.killed) {
+          return;
+        }
+
+        if (this.closed && this[kDocuments].length === 0) {
+          return;
+        }
+
+        if (this[kId] != null && this.isDead && this[kDocuments].length === 0) {
+          return;
+        }
```

Comment:
Yes, and ty! I intended these to be written as if you would read them:
- if killed - return
- if closed AND there's no more documents (because you can be closed but still finishing the last batch) - return
- if there is a truthy Id (which means we already ran the initial command) the cursor isDead (generally, id = 0) and there are no more documents - return